src/wechaty/accessory.py

```diff
@@ -25,18 +25,22 @@
     # overload,
     # cast,
     Optional,
-)
-from wechaty_puppet import get_logger   # type: ignore
+    TypeVar, Generic)
+
+from wechaty.exceptions import WechatyAccessoryBindingError
+from wechaty_puppet import get_logger  # type: ignore
 
 # pylint:disable=R0401
 if TYPE_CHECKING:
-    from wechaty_puppet import Puppet   # type: ignore
+    from wechaty_puppet import Puppet  # type: ignore
     from .wechaty import Wechaty
 
 log = get_logger('Accessory')
 
+PayloadType = TypeVar('PayloadType')
+
 
-class Accessory:
+class Accessory(Generic[PayloadType]):
     """
     Translate the function from TypeScript to Python
     See: https://github.com/wechaty/wechaty/blob/master/src/accessory.ts
```

Comment:
In order to refactor the `is_ready` function and `puppet` property, you add the generic type to the `Accessory` abstract type, and all of the implementation subclass, eg: `Contact`, `Room`, `Friendship` ..., must add the Payload Type defination to the top of the class.  I think this `refactor desgin` is not so efficient:  `two class methods` vs `class generic class type`. It doesn't win a lot. 

On the another hand, below code will make me doubt at semantic aspect:
```python
class Contact(Accessory[ContactPayload], AsyncIOEventEmitter):
    pass
```
And more, `Payload` Type should be the internal defination, which shuould not be exposed to user.  In your design, `payload` type is more revealing. We should keep the module as clean as the [ts-wechaty](https://github.com/wechaty/wechaty/blob/master/src/user/room.ts). 

How do you think about it ?

Reply:
I didn't look deep into the ts version before I think about this refactoring, I just think that the `ready()`, `is_ready()` method and `payload` property is somehow duplicated implemented through all of the Accessory classes, we might not have to repeat it. But when I read about the relating ts version, I see they are just a interpretion of them, and keep the mapping from the ts version is not bad anyway.

Reply:
But if we change the sight back to the python version as an independent as I did, I think throwing away the duplicated method is considerable. As a trade off thinking, we list the pros and cons for the refactoring work here:

**PROS:**

* We can throw away those methods and properties and all the relating processing code throughout the many classes, which are somehow likely but maybe a bit different betweent them.
* When we consider to define a new or maintaining or an existing `Accessory` derived class, we don't need to take care about those treatment and have a strong believe that their behaviors are same.

**CONS:**

* We added a Generic Type feature to the code, which may cause a performance loss.
* We kick the code style out of the TS version trace, which may make it harder to compare in the future.

For more, **I don't think** the Payload type is exposed to the user, it is still wrapped inside our wechaty package, no more exposion than before, and the user is not expect to use the Accessory class directly, so I think we don't have to worry about your consideration about the exposion issue.

---

For those overall consideration, I think the refactoring still have improvement anyway.

---

Howerver, I don't have enough confidence for the above thinking of mine because of my lack of experience. I'm not sure whether it is a good practise or whether did I trapped into an anti-pattern.

So I think it would be nice if more python master can review the design together and share the best practise for the case.

Reply:
We're glad you've started thinking about `python-wechaty` architecture, which means you have proven to be a highly qualified contributor to community. 

Although `payload` is the attribute that every subclass of `Accessory` will maintained, but the type of it is different which means we should not wrap it to one place. We reduce one attribute (actual one attribute and one publich method `is_ready` ) but also add generic type definition to the subclass.

I have to say that you make a greate design for `python-wechaty`.   @huan How do you think about it ?

Reply:
I think it's ok for `python-wechaty` code refactor.  Please make the ci green and request review.